src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -1205,41 +1234,81 @@ private boolean isSuspended()
             private int srcSeqNumStart = -1;
 
             /**
+             * Return whether or not the current frame is being accepted.
+             * srcSeqNumLimit would not be updated (remain as -1) if the
+             * frame were either:
+             * a) an old frame of an adaptive stream OR
+             * b) a frame from an adaptive stream on a layer we're not using
+             * @return true if we're accepting this frame
+             */
+            private boolean isAccepting()
+            {
+                return srcSeqNumLimit != -1;
+            }
+
+            /**
+             * Update this {@link SeenFrame} with information from its
+             * corresponding {@link FrameDesc}
+             * @param frame the {@link FrameDesc} from which to update
+             */
+            void update(FrameDesc frame)
+            {
+                if (this == mostRecentSentFrame /* the max frame can expand */
+                    || availableQualityIndices == null
+                    || availableQualityIndices.length < 2)
+                {
+                    if (frame.lastSequenceNumberKnown())
+                    {
+                        srcSeqNumLimit = frame.getEnd();
+                    }
+                    else
+                    {
+                        srcSeqNumLimit = frame.getMaxSeen();
+                    }
+                }
+                if (isAccepting())
+                {
+                    if (frame.firstSequenceNumberKnown())
+                    {
+                        srcSeqNumStart = frame.getStart();
+                    }
+                    else
+                    {
+                        srcSeqNumStart = frame.getMinSeen();
+                    }
+                }
+            }
+
+            /**
              * Defines an RTP packet filter that controls which packets to be
              * written into some arbitrary target/receiver that owns this
-             * {@link SeenFrame}.
+             * {@link SeenFrame}.  {@link SeenFrame#update(FrameDesc)} should
+             * be called with the {@link FrameDesc} for this packet before
+             * calling accept.
              *
-             * @param source the {@link FrameDesc} that the RTP packet belongs
-             * to.
              * @param pkt the packet for which to decide to accept
              * @return <tt>true</tt> to allow the specified packet/<tt>buffer</tt>
              * to be written into the arbitrary target/receiver that owns this
              * {@link SeenFrame} ; otherwise, <tt>false</tt>
              */
-            boolean accept(FrameDesc source, RawPacket pkt)
+            boolean accept(RawPacket pkt)
```

Comment:
As someone with really no context here, isAccepting and accept (shouldAccept) existing together kind of confuses me. They seem to do the same thing based on name.